ogzaf/demographics.py

```diff
@@ -1,96 +1,352 @@
+        end_year (int): end year of the data
+        download (bool): whether to download the data from the UN Data Portal.
+            If False, a path must be specified in the path_folder argument.
+        path_folder (None or str): string path to folder where data are stored
 
     Returns:
-        df (Pandas DataFrame): DataFrame of UN data
+        fert_rates_df (DataFrame): dataset with fertility rates by age
     """
-    target = (
-        "https://population.un.org/dataportalapi/api/v1/data/indicators/"
-        + variable_code
-        + "/locations/"
-        + country_id
-        + "/start/"
-        + str(start_year)
-        + "/end/"
-        + str(end_year)
-    )
-
-    # get data from url
-    response = requests.get(target)
-    # Converts call into JSON
-    j = response.json()
-    # Convert JSON into a pandas DataFrame.
-    # pd.json_normalize flattens the JSON to accomodate nested lists
-    # within the JSON structure
-    df = pd.json_normalize(j["data"])
-    # Loop until there are new pages with data
-    while j["nextPage"] is not None:
-        # Reset the target to the next page
-        target = j["nextPage"]
-        # call the API for the next page
-        response = requests.get(target)
-        # Convert response to JSON format
-        j = response.json()
-        # Store the next page in a data frame
-        df_temp = pd.json_normalize(j["data"])
-        # Append next page to the data frame
-        df = df.append(df_temp)
-
-    return df
+    if end_year is None:
+        end_year = start_year
+    # UN variable code for Population by 1-year age groups and sex
+    pop_code = "47"
+    # UN variable code for Fertility rates by age of mother (1-year)
+    fert_code = "68"
+
+    if download:
+        pop_target = (
+            "https://population.un.org/dataportalapi/api/v1/data/indicators/"
+            + pop_code
+            + "/locations/"
+            + country_id
+            + "/start/"
+            + str(start_year)
+            + "/end/"
+            + str(end_year)
+            + "?format=csv"
+        )
+        fert_target = (
+            "https://population.un.org/dataportalapi/api/v1/data/indicators/"
+            + fert_code
+            + "/locations/"
+            + country_id
+            + "/start/"
+            + str(start_year)
+            + "/end/"
+            + str(end_year)
+            + "?format=csv"
+        )
+    else:
+        pop_target = os.path.join(DATA_DIR, "un_zaf_pop.csv")
+        fert_target = os.path.join(DATA_DIR, "un_zaf_fert.csv")
+
+    # Convert .csv file to Pandas DataFrame
+    pop_df = pd.read_csv(
+        pop_target,
+        sep="|",
+        header=1,
+        usecols=["TimeLabel", "SexId", "Sex", "AgeMid", "Value"],
+        float_precision="round_trip",
+    )
+    fert_rates_df = pd.read_csv(
+        fert_target,
+        sep="|",
+        header=1,
+        usecols=["TimeLabel", "AgeMid", "Value"],
```

Comment:
```suggestion
        usecols=["TimeLabel", "AgeStart", "Value"],
```
The "AgeMid" is 1 for ages 0 and 1. This may be the reason for the issue in line 123